src/guessUnicodePunctuation.js

```diff
@@ -39,6 +39,8 @@ function guessUnicodePunctuation() {
 		[/'/g, '’'], // ... and finally the apostrophes should be remaining
 		[/(?<!\.)\.{3}(?!\.)/g, '…'], // horizontal ellipsis (but not more than three dots)
 		[/ - /g, ' – '], // en dash as separator
+		[/(\d{4})-(\d{2})-(\d{2})/g, '$1‐$2‐$3'], // hyphens for ISO 8601 dates, e.g. 1987‐07–30
+		[/(\d{4})-(\d{2})/g, '$1‐$2'], // hyphen for ISO 8601 partial dates, e.g. 2016-04
 		[/(\d+)-(\d+)/g, '$1–$2'], // en dash for ranges where it means "to", e.g. 1965–1972
```

Comment:
This should work for all full or partial ISO 8601 dates but I would have preferred a more compact solution that only requires one additional regex search:
```suggestion
		[/(\d{4})-(\d{2})/g, '$1‐$2'], // hyphen for ISO 8601 dates (YYYY-MM-DD), e.g. 2016-04
		[/(?<!‐)(\d+)-(\d+)/g, '$1–$2'], // en dash for ranges, e.g. 1965–1972 (but not the MM‐DD part of dates)
```
Unfortunately this specific negative lookbehind for a hyphen does not work, it ignores the leading hyphen before MM-DD (that was replaced by the previous regex) nevertheless and treats it as a range 😕

@jesus2099 Do you have an idea why this does not work as expected?